### DISCUSSION TITLE
http taking prevalence over https

### DISCUSSION THREAD

**Discussion by digityne** [PiranhaCMS/piranha.core#1639] [Q&A] [answered]
Hi all, I'm a little confused about why this is happening, it seems to be something to do with my piranha mvc project setup but I don't know why or how to fix it - I'm relatively new to web tech so please bear with me!

Normally I deploy my asp.net core sites using IIS on a windows VM. I have bindings for both `http` port 80 and `https` port 443.
When I then visit a domain name pointing to the VM e.g. `www.xyz.com` it always resolves to `https`, even if I prefix the url with `http` it changes automatically.

This isn't happening for my piranha project, and if I decorate the CMSController with `RequireHttps` it works on some pages, but not on the `manager` pages.

Any idea why this is happening? I've tried a normal asp.net mvc project on the same domain name and it works as expected, but using the default piranha project seems to have some different settings.

**Comment by tidyui** [comment, marked as answer]
Most templates provided by Microsoft contains the following line in the `Configure` method in `Startup.cs` which will effectively redirect all requests to `http` to `https`. However our templates do not contain this by default, have you tried adding this line manually?

~~~
app.UseHttpsRedirection();
~~~

Regards

**Reply by digityne** [reply]
Man I feel so silly, I had been googling this, scouring my project properties for anything to do with SSL and HTTPS, changing the controller, comparing projects and I even did look at `Startup.cs` in another project which has the `UseHttpsRedirection` method - I just overlooked it.

Thanks a tonne, I won't forget this issue any time soon.
